arangod/Aql/DistributeExecutor.cpp

```diff
@@ -224,6 +232,9 @@ std::pair<ExecutionState, bool> ExecutionBlockImpl<DistributeExecutor>::getBlock
 
   while (buf.size() < atMost) {
     if (_index == _buffer.size()) {
+      if (getQuery().killed()) {
+        THROW_ARANGO_EXCEPTION(TRI_ERROR_QUERY_KILLED);
+      }
```

Comment:
seems expensive to check this inside the loop? shouldn't this be checkd on the DBServers as well

Reply:
The cost should be negligible here, as the check is only performed when a new block is fetched anyway, but not for every "row".